### ISSUE TITLE
[Feature request] Option for quick system buttons on bottom left

### ISSUE THREAD

**Issue by BinaryQuantumSoul** [BinaryQuantumSoul/sdnext-modernui#27]
Copy behavior of those buttons
![image](https://github.com/BinaryQuantumSoul/sdnext-modernui/assets/19613657/2b1499d2-71ad-41fe-9aaf-2a0168ff80da)

Next to this one:
![image](https://github.com/BinaryQuantumSoul/sdnext-modernui/assets/19613657/97139ac3-0d3e-4a6b-8575-6beedb13bcc4)

**Comment by vladmandic** [comment]
i'm not sure i agree with this one.

**Comment by BinaryQuantumSoul** [comment]
I know you don't but I was told by Aptro that some users might find it very useful, so we could set it as option

**Comment by vladmandic** [comment]
sure, if there is a use case for each of those buttons.

**Comment by Aptronymist** [comment]
Yeah, I would be behind this 💯 as would many others, while I know vlad doesn't dig the whole 'unload' and 'reload' thing, and I do agree bug reports should be filed for things causing sdnext to misbehave and spit errors, it's insanely useful for when things do screw up, you really have little other recourse to solving the issue other than loading another model and then switching back, same thing ultimately. 

Better to have a fix and not need it than to not have a fix and people are irritated by the bugs even more.
@vladmandic Do you think some kind of detection for the "didn't unapply" errors could be put in? Doing a model reload automatically would solve a lot of headaches when those things are going wrong.

**Comment by Aptronymist** [comment]
Though now that I think about it, either putting in a "Report Error" button that would dump all relevant diagnostic info as well as a chunk of the log going back to the generation or two before the error started (at least all changes in settings used to create it) until the point when the button was pushed might be damn handy, or we could revisit the automatic error reporting, I think the way I devised would be simplicity as it just scans the previous log on startup for errors and then files them after anonymizing it.

**Comment by vladmandic** [comment]
> Though now that I think about it, either putting in a "Report Error" button that would dump all relevant diagnostic info as well as a chunk of the log going back to the generation or two before the error started (at least all changes in settings used to create it) until the point when the button was pushed might be damn handy, or we could revisit the automatic error reporting, I think the way I devised would be simplicity as it just scans the previous log on startup for errors and then files them after anonymizing it.

that is a completely different topic. and i'm not doing anything that would ease and/or automate issue creation until there are people actually working on issues - right now i'm handling 90% of them.

**Comment by Aptronymist** [comment]
> > Though now that I think about it, either putting in a "Report Error" button that would dump all relevant diagnostic info as well as a chunk of the log going back to the generation or two before the error started (at least all changes in settings used to create it) until the point when the button was pushed might be damn handy, or we could revisit the automatic error reporting, I think the way I devised would be simplicity as it just scans the previous log on startup for errors and then files them after anonymizing it.
> 
> that is a completely different topic. and i'm not doing anything that would ease and/or automate issue creation until there are people actually working on issues - right now i'm handling 90% of them. and pruning logs and all the stuff needed for that is a rathole i'm not doing. prs are welcome.

Of course, but they are mitigation strategies assuming we did have some sort of autofix. Can discuss whenever. I'll look at a log scanning script.

**Comment by BinaryQuantumSoul** [comment]
> Yeah, I would be behind this 💯 as would many others, while I know vlad doesn't dig the whole 'unload' and 'reload' thing, and I do agree bug reports should be filed for things causing sdnext to misbehave and spit errors, it's insanely useful for when things do screw up, you really have little other recourse to solving the issue other than loading another model and then switching back, same thing ultimately. There's zero reason to make restarting and shutting down a hassle to get to if we have the ui space.
> 
> Better to have a fix and not need it than to not have a fix and people are irritated by the bugs even more.
> @vladmandic Do you think some kind of detection for the "didn't unapply" errors could be put in? Doing a model reload automatically would solve a lot of headaches when those things are going wrong.

So we need (five) new buttons on bottom of right bar, next to expand arrows ? Optional ?

**Comment by brknsoul** [comment]
I don't think we need all 5 of them, Restart and Shutdown should be sufficient, unless people are constantly unloading and reloading checkpoints?

This might be over-engineering it, but an "are you sure" flyout would be nice, to prevent accidental clicking when opening/closing the sidebar.

![image](https://github.com/BinaryQuantumSoul/sdnext-modernui/assets/1969381/91a24a70-362f-48fd-bcc4-daadea480957)

**Comment by BinaryQuantumSoul** [comment]
The 3rd and 4th points of this proposal https://github.com/BinaryQuantumSoul/sdnext-modernui/issues/54#issue-2310324518 should be done anyway imo. So I think we need to remove the collapse buttons and instead have tabs work like normal tabs: expanded when highlighted, collapsed when not. Therefore there would not be the need for a confirmation, though there still will be the full-expand button

**Comment by Aptronymist** [comment]
> I don't think we need all 5 of them, Restart and Shutdown should be sufficient, unless people are constantly unloading and reloading checkpoints?

Normally I'd agree, but it is the **best fix for many errors that pop up, which leads to a lot of frustration** so I think making it **more accessible and making it clear when to use it and why** would be of benefit. Some expansion of the Troubleshooting wiki page would do nicely, assuming we can all agree on what those steps should be and why.

Not to mention when I'm using two AI applications (often these days) at the same time I find I need to dump a model out of memory of SDNext (bloated SDXL!) so there's room for the other app to have a model loaded and do work, particularly where that might not be so easy to do, if at all, in that other application itself.

I use it a lot, and from what I understand several other henchmen do as well for various reasons.

**Comment by BinaryQuantumSoul** [comment]
It's already available in two clicks in system tab, so I don't know if that should be default. But an option seems reasonable
